### PR TITLE
apply pre-commit formats

### PR DESCRIPTION
**Pull request by fneum**

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -37,6 +37,10 @@ PyPSA: Python for Power System Analysis
     :target: https://gitter.im/PyPSA/community?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge
     :alt: Chat on Gitter
 
+.. image:: https://results.pre-commit.ci/badge/github/PyPSA/PyPSA/master.svg
```
Comment (euronion) [inline doc/index.rst:40]: `black` badge missing here

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -24,13 +24,13 @@ repos:
...
-    args: ['--ignore-regex="\b[A-Z]+\b"'] # Ignore capital case words, e.g. country codes
-    types_or: [python, rst, markdown]
-    files: ^(scripts|doc)/
+# - repo: https://github.com/codespell-project/codespell
+#   rev: v2.1.0
+#   hooks:
+#   - id: codespell
+#     args: ['--ignore-regex="\b[A-Z]+\b"'] # Ignore capital case words, e.g. country codes
+#     types_or: [python, rst, markdown]
+#     files: ^(scripts|doc)/
```
Comment (euronion) [inline .pre-commit-config.yaml:33]: Why exclude codespell?

**Review by euronion — CHANGES_REQUESTED** [review]
I'll not check the automatic changes and trust that they are fine ;-)

File: .pre-commit-config.yaml (hunk 2 above)
Comment (fneum) [reply .pre-commit-config.yaml:33]: It suggested some corrections for URLs which I couldn't ignore.

**Comment by FabianHofmann** [comment]
But now GH will think that @fneum did all the changes right? Initial idea was to let pre-commit CI do the large parts. Or does that not work for a reason?

**Comment by fneum** [comment]
Ah, yes. Happy to do it another way. I thought someone had to apply it first.

**Pull request merged**